pkg/scraper/client/resource/decode.go

```diff
@@ -0,0 +1,158 @@
+	node.CumulativeCpuUsed = uint64(sample.Value * 1e9)
+	if sample.Timestamp != 0 {
+		//unit of timestamp is millisecond, need to convert to nanosecond
+		node.Timestamp = time.Unix(0, int64(sample.Timestamp*1e6))
+	}
+}
+
+func parseNodeMemUsageMetrics(sample *model.Sample, node *storage.MetricsPoint) {
+	node.MemoryUsage = uint64(sample.Value)
+	if node.Timestamp.IsZero() && sample.Timestamp != 0 {
+		//unit of timestamp is millisecond, need to convert to nanosecond
+		node.Timestamp = time.Unix(0, int64(sample.Timestamp*1e6))
+	}
+}
+
+func parseContainerCpuMetrics(sample *model.Sample, pods map[apitypes.NamespacedName]storage.PodMetricsPoint) {
+	namespaceName := getNamespaceName(sample)
+	containerName := string(sample.Metric[containerNameMetricName])
+	if _, findPod := pods[namespaceName]; !findPod {
+		pods[namespaceName] = storage.PodMetricsPoint{Containers: make(map[string]storage.MetricsPoint)}
+	}
+	if _, findContainer := pods[namespaceName].Containers[containerName]; !findContainer {
+		pods[namespaceName].Containers[containerName] = storage.MetricsPoint{}
+	}
+	//unit of node_cpu_usage_seconds_total is second, need to convert to nanosecond
+	containerMetrics := pods[namespaceName].Containers[containerName]
+	containerMetrics.CumulativeCpuUsed = uint64(sample.Value * 1e9)
+	if sample.Timestamp != 0 {
+		//unit of timestamp is millisecond, need to convert to nanosecond
+		containerMetrics.Timestamp = time.Unix(0, int64(sample.Timestamp*1e6))
+	}
+	pods[namespaceName].Containers[containerName] = containerMetrics
+}
+
+func parseContainerMemMetrics(sample *model.Sample, pods map[apitypes.NamespacedName]storage.PodMetricsPoint) {
+	namespaceName := getNamespaceName(sample)
+	containerName := string(sample.Metric[containerNameMetricName])
+
+	if _, findPod := pods[namespaceName]; !findPod {
+		pods[namespaceName] = storage.PodMetricsPoint{Containers: make(map[string]storage.MetricsPoint)}
+	}
+	if _, findContainer := pods[namespaceName].Containers[containerName]; !findContainer {
+		pods[namespaceName].Containers[containerName] = storage.MetricsPoint{}
+	}
+	containerMetrics := pods[namespaceName].Containers[containerName]
+	containerMetrics.MemoryUsage = uint64(sample.Value)
+	if containerMetrics.Timestamp.IsZero() && sample.Timestamp != 0 {
+		//unit of timestamp is millisecond, need to convert to nanosecond
+		containerMetrics.Timestamp = time.Unix(0, int64(sample.Timestamp*1e6))
+	}
+	pods[namespaceName].Containers[containerName] = containerMetrics
+}
+
+func checkContainerMetricsTimestamp(podMetric storage.PodMetricsPoint) map[string]storage.MetricsPoint {
+	podMetrics := make(map[string]storage.MetricsPoint)
+	for containerName, containerMetric := range podMetric.Containers {
+		if containerMetric != (storage.MetricsPoint{}) {
+			//drop metrics when Timestamp is zero
+			if containerMetric.Timestamp.IsZero() || containerMetric.CumulativeCpuUsed == 0 || containerMetric.MemoryUsage == 0 {
```

Comment:
I am wondering if it is really meaningful to check for `containerMetric.Timestamp.IsZero()`. In practice, this should never happen since Prometheus metrics created via the official client always have a timestamp.